src/vopr/main.go

```diff
@@ -0,0 +1,766 @@
+	commit_string := hex.EncodeToString(message.commit[:])
+
+	file_name := ""
+	if message.bug == 1 || message.bug == 2 {
+		file_name = fmt.Sprintf(
+			"%s/%d_%d_%s",
+			issue_directory,
+			message.bug,
+			message.seed,
+			commit_string,
+		)
+	} else if message.bug == 3 {
+		file_name = fmt.Sprintf(
+			"%s/%d_%s_%s",
+			issue_directory,
+			message.bug,
+			commit_string,
+			stack_trace_hash,
+		)
+	}
+	return file_name
+}
+
+// Fetch available branches from GitHub and checkout the correct commit if it exists.
+func checkout_commit(commit string, message_hash []byte) error {
+	// Git commands need to be run with the TigerBeetle directory as their working_directory
+	fetch_command := exec.Command("git", "fetch", "--all")
+	fetch_command.Dir = tigerbeetle_directory
+	error := fetch_command.Run()
+	if error != nil {
+		error_message := fmt.Sprintf("Failed to run git fetch: %s", error.Error())
+		log_error(error_message, message_hash)
+		return error
+	}
+
+	// Checkout the commit specified in the vopr_message
+	checkout_command := exec.Command("git", "checkout", commit)
+	checkout_command.Dir = tigerbeetle_directory
+	error = checkout_command.Run()
+	if error != nil {
+		error_message := fmt.Sprintf("Failed to run git checkout: %s", error.Error())
+		log_error(error_message, message_hash)
+		return error
+	}
+
+	// Inspect the git logs.
+	log_command := exec.Command("git", "log")
+	log_command.Dir = tigerbeetle_directory
+	log_output := make([]byte, 47)
+	log_output, error = log_command.Output()
+	if error != nil {
+		error_message := fmt.Sprintf("Failed to run git log: %s", error.Error())
+		log_error(error_message, message_hash)
+		return error
+	}
+
+	// Check the log to determine if the commit has been successfully checked out.
+	current_commit := string(log_output[0:47])
+	checkout_successful, error := regexp.MatchString("^commit "+commit, current_commit)
+	if error != nil {
+		error_message := fmt.Sprintf(
+			"Regular expression failure while checking the git logs: %s",
+			error.Error(),
+		)
+		log_error(error_message, message_hash)
+		return error
+	}
+
+	if !checkout_successful {
+		error = fmt.Errorf("Checkout failed")
+		return error
+	}
+
+	return nil
+}
+
+// The Vopr is run from the TigerBeetle directory and its output is captured.
+func run_vopr(seed uint64, output *vopr_output, message_hash []byte) {
+	// Create a limited_buffer to read the vopr output
+	var vopr_std_err_buffer bytes.Buffer
+	limited_buffer := size_limited_buffer{
+		buffer:       &vopr_std_err_buffer,
+		byte_budget:  int(max_length_of_vopr_output),
+		size_reached: make(chan bool),
+	}
+
+	// The channel monitors if the Vopr completes before the maximum output is reached.
+	vopr_completed := make(chan bool)
+
+	vopr_path := tigerbeetle_directory + "/scripts/vopr.sh"
+
+	// Runs in debug mode
+	log_info("Running the Vopr...", message_hash)
+	cmd := exec.Command(
+		"/bin/bash",
+		vopr_path,
+		fmt.Sprint(seed),
```

Comment:
Is there a way to Sprint this explicitly as an integer? The reason I ask is that the risk with executing commands is that an attacker can try and pass hostile input to inject into the command and so run their own command on the system to take it over or steal data. The seed is already a `uint64` here but it would be good to make this explicit right here at the call site.